### PR TITLE
Thread transport dependencies

### PR DESCRIPTION
**Pull request by kriskowal**
Paving the way for the tracing dependency.

### CODE REVIEW

File: transport/dependencies.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package transport
+
+// Dependencies is the interface of any object useful for passing injected
+// dependencies into inbound and outbound transports.
+type Dependencies interface {
```
Comment (breerly) [inline transport/dependencies.go]: Suggest shortening to `Deps` - it's still clear what the object is at this length.

File: transport/tchannel/inbound.go [hunk 2]
```diff
@@ -69,7 +69,7 @@ func (i *inbound) Channel() *tchannel.Channel {
 	return i.ch
 }
 
-func (i *inbound) Start(h transport.Handler) error {
+func (i *inbound) Start(h transport.Handler, d transport.Dependencies) error {
```
Comment (breerly) [inline transport/tchannel/inbound.go]: @abhinav @kriskowal what do you think about putting the handler on `d`?

**Comment by breerly** [comment]
LGTM to me, but I think we should consider combining `h` and `d`.

File: transport/tchannel/inbound.go (hunk 2 above)
Comment (abhinav) [inline transport/tchannel/inbound.go]: Handler is an Inbound-only dependency. I don't think we should put it on Dependencies. It adds yet another thing that can be nillable in many cases.

**Pull request merged**